tests/tracer/test_uwsgi.py

```diff
@@ -0,0 +1,179 @@
+try:
+    import asyncio
+    from tests.contrib.asyncio.utils import mark_asyncio
+except ImportError:
+    asyncio = None
+
+    def mark_asyncio(f):
+        return f
+
+
+from collections import defaultdict
+import os
+import re
+import subprocess
+import sys
+import textwrap
+import time
+
+import pytest
+
+from ddtrace.vendor import six
+
+from ddtrace.compat import httplib, yield_from_coro
+from tests import snapshot
+
+uwsgi_app = os.path.join(os.path.dirname(__file__), "uwsgi-app.py")
+
+HTTP_PORT = 9091
+
+
+@pytest.fixture
+def event_loop():
+    # Start a new event loop as loop is closed by @mark_asyncio
+    loop = asyncio.new_event_loop()
+    asyncio.set_event_loop(loop)
+    yield loop
+
+
+@pytest.fixture
+def uwsgi():
+    cmd = ["uwsgi", "--need-app", "--die-on-term", "--http-socket", ":{}".format(HTTP_PORT), "--wsgi-file", uwsgi_app]
+
+    def _run_uwsgi(*args):
+        env = os.environ.copy()
+        if sys.version_info[0] == 2:
+            # On Python2, it's impossible to import uwsgidecorators without this hack
+            env["PYTHONPATH"] = os.path.join(
+                os.environ.get("VIRTUAL_ENV", ""),
+                "lib",
+                "python%s.%s" % (sys.version_info[0], sys.version_info[1]),
+                "site-packages",
+            )
+
+        return subprocess.Popen(cmd + list(args), stdout=subprocess.PIPE, stderr=subprocess.STDOUT, env=env)
+
+    yield _run_uwsgi
+
+
+@snapshot(async_mode=False)
+def test_uwsgi_single_request(uwsgi):
+    proc = uwsgi("--enable-threads", "--processes", "1", "--master")
+    # Give some time to child to actually startup
+    time.sleep(3)
```

Comment:
This is going to be flaky. You need to check the output as I did in the profiler cases.

Reply:
This is still true. If your code below is solid and works 100% of the case (which I hope) then just go ahead and remove the sleep.